clients/src/main/java/org/apache/kafka/common/security/auth/KafkaPrincipal.java

```diff
@@ -22,6 +22,10 @@
 public class KafkaPrincipal implements Principal {
     private final String name;
 
+    public static KafkaPrincipal anonymousPrincipal() {
+        return new KafkaPrincipal("ANONYMOUS");
```

Comment:
Excuse me if my question is really naive/stupid, but there's a need to return a **new** anonymous object each time this function is called? I mean, KafkaPrincipal is a placeholder and the principal is anonymous, so why not have:
```
public class KafkaPrincipal implements Principal {
     private final String name;
     public static final KafkaPrincipal ANONYMOUS = new KafkaPrincipal("ANONYMOUS");
(...)
```
Besides saving on object creation (kinda of Flyweight pattern), we could replace the ``anonymousPrincipal()`` method by ``KafkaPrincipal.ANONYMOUS``. Does it make sense?